src/Control/Bind.purs

```diff
@@ -63,6 +63,17 @@ infixr 1 bindFlipped as =<<
 instance bindFn :: Bind ((->) r) where
   bind m f x = f (m x) x
 
+-- | The `Array` monad works like a nested for loop. Each
```

Comment:
Sorry, just to clarify what I was trying to get at before: I think it's best to provide a precise explanation which says what the thing does before trying to build intuition. In this case I think we should follow the lead of the docs in Data.Array.concatMap to start with before mentioning the similarity to a nested for loop. Perhaps:

> The `bind`/`>>=` function for Array works by applying a function to each element in the array, and flattening the results into a single, new array. Array's `bind` works like a nested for loop; each `bind` adds another level of nesting in the loop:

Reply:
Makes sense. I'll use that rendering.